test/puppetlabs/puppetdb/config_test.clj

```diff
@@ -173,6 +169,15 @@
                      (.mkdir)
                      (.setWritable true))]
       (is (= (validate-vardir (vardir filename))
+             (vardir filename)))))
+
+  (testing "should support relative paths"
+    (let [filename "./target/totally/okay"]
+      (doto (fs/file filename)
+        (.deleteOnExit)
+        (.mkdirs)
```

Comment:
I think that anything that begins with a leading slash is considered an absolute path by the various java/clojure APIs.